src/cli/index.ts

```diff
@@ -0,0 +1,114 @@
+const yargs = require('yargs'); // eslint-disable-line
+const { hideBin } = require('yargs/helpers'); // eslint-disable-line
+import { array, either, taskEither } from 'fp-ts';
+import { serialize as serializeOpenAPI3 } from '../language/typescript/3.0';
+import { generate, Language } from '../index';
+import { Decoder } from 'io-ts';
+import { Reader } from 'fp-ts/lib/Reader';
+import { ResolveRefContext } from '../utils/ref';
+import { TaskEither } from 'fp-ts/lib/TaskEither';
+import * as $RefParser from 'json-schema-ref-parser';
+import * as path from 'path';
+import { pipe } from 'fp-ts/lib/pipeable';
+import { OpenapiObject, OpenapiObjectCodec } from '../schema/3.0/openapi-object';
+import { SwaggerObject } from '../schema/2.0/swagger-object';
+import { serialize as serializeSwagger } from '../language/typescript/2.0';
+import { AsyncAPIObject, AsyncAPIObjectCodec } from '../schema/asyncapi-2.0.0/asyncapi-object';
+import { serialize as serializeAsyncApi } from '../language/typescript/asyncapi-2.0.0';
+
+const options = yargs(hideBin(process.argv))
+	.usage('swagger-codegen-ts -o <output_path> [flags] <path/to/spec.yaml>...')
+	.option('baseDir', {
+		alias: 'b',
+		describe: 'base directory for <specs> and <outDir> paths',
```

Comment:
It's correct if I understand you correctly
`outDir` is relative to `baseDir`

Reply:
After thinking a bit more on this, probably making `outDir` relative to the `baseDir` does not make much sense.
In the new commit this will be like this:

- `outDir` is relative to the cwd (unless an absolute path is given);
- each `spec` is relative to `baseDir` if it is provided, otherwise relative to the cwd